Include/Beman/InplaceVector/inplace_vector.hpp

```diff
@@ -0,0 +1,755 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <compare>
+#include <concepts>
+#include <iterator>
+#include <limits>
+#include <memory>
+#include <new>
+#include <ranges>
+#include <type_traits>
+
+namespace Beman::InplaceVector {
+namespace detail {
+// Exposition-only container-compatible-range
+template <typename T>
+concept container_compatible_range_impl = requires(T &&t) {
+  {
+    std::ranges::begin(t)
+  } -> std::same_as<typename std::remove_reference_t<T>::iterator>;
+  {
+    std::ranges::end(t)
+  } -> std::same_as<typename std::remove_reference_t<T>::iterator>;
+};
+} // namespace detail
+template <typename T>
+constexpr bool container_compatible_range =
+    detail::container_compatible_range_impl<T>;
+
+template <bool Condition, typename TrueType, typename FalseType>
+using If = typename std::conditional<Condition, TrueType, FalseType>::type;
+
+// The internal size type can be smaller than std::size_t when capacity allows
+// for it.
+template <std::size_t Capacity>
+using inplace_vector_internal_size_type =
+    If<Capacity <= std::numeric_limits<uint8_t>::max(), uint8_t,
+       If<Capacity <= std::numeric_limits<uint16_t>::max(), uint16_t,
+          If<Capacity <= std::numeric_limits<uint32_t>::max(), uint32_t,
+             uint64_t>>>;
+
+//  Base class for inplace_vector
+template <typename T, std::size_t Capacity>
+struct inplace_vector_destruct_base {
+  using size_type = std::size_t;
+  using internal_size_type = inplace_vector_internal_size_type<Capacity>;
+
+  alignas(T) unsigned char elems[Capacity * sizeof(T)] = {};
```

Comment:
something I notice on the second pass -- In this paper at least the recommendation is to use std::byte with alignas -- I'm guessing unsigned char is the same, but I'm not alignment expert. I don't want to hold the PR for this, but curious if the better practice now is byte.

https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p1413r3.pdf